test/SlowTests/Issues/RavenDB_17046.cs

```diff
@@ -0,0 +1,187 @@
+        {
+            var result = Act<dynamic>("@all_docs", "c");
+
+            Assert.Equal(result.Count, 2 * CollectionSize);
+        }
+
+        private List<T> Act<T>(string collection, string startsWithParam)
+        {
+            string fullStatement = $"from {collection} where startsWith(id(),'{startsWithParam}')";
+            List<T> wholeCollectionByPaging;
+
+            using (var store = GetDocumentStore())
+            {
+                PrepareDatabaseForTest(store);
+                using (var session = store.OpenSession())
+                {
+                    wholeCollectionByPaging = new List<T>();
+                    IList<T> results;
+                    int pageNumber = 0;
+                    int pageSize = 101;
+                    int skippedResults = 0;
+                    do
+                    {
+                        results = session
+                            .Advanced
+                            .RawQuery<T>(fullStatement)
+                            .Statistics(out QueryStatistics stats)
+                            .Skip((pageNumber * pageSize) + skippedResults)
+                            .Take(pageSize)
+                            .ToList();
+                        wholeCollectionByPaging.AddRange(results);
+                        skippedResults += stats.SkippedResults;
+                        pageNumber++;
+
+                        if (results.Count == pageSize)
+                        {
+                            Assert.Equal(-1, stats.TotalResults);
+                            Assert.Equal(-1, stats.LongTotalResults);
+                        }
+                    }
+                    while (results.Count > 0);
+                }
+            }
+
+            return wholeCollectionByPaging;
+        }
+
+        private void PrepareDatabaseForTest(DocumentStore store)
+        {
+            using (var bulkInsert = store.BulkInsert())
+            {
+                for (int i = 0; i < CollectionSize; ++i)
+                {
+                    bulkInsert.Store(_categories[i]);
+                    bulkInsert.Store(_companies[i]);
+                }
+            }
+        }
+        private static int CompaniesSortById(Company a, Company b)
+        {
+            return a.Id.CompareTo(b.Id);
+        }
+    }
+
+    public class RavenDB_17046_LowLevelTests : RavenLowLevelTestBase
+    {
+        private DocumentDatabase _documentDatabase;
+        private IDisposable _disposeDatabase;
+
+        public RavenDB_17046_LowLevelTests(ITestOutputHelper output) : base(output)
+        {
+        }
+        
+        public void Initialize([CallerMemberName] string caller = null)
+        {
+            _disposeDatabase = CreatePersistentDocumentDatabase(NewDataPath(prefix: "DocumentsCrud"), out _documentDatabase, caller);
+        }
+
+        [Fact]
+        public void CheckIfNotMatchedItemCountIntoSkippedValues()
+        {
+            Initialize();
```

Comment:
Minor (no need to make any code changes) - you have only one test here. It would be simpler to just put:

`using (var db = CreatePersistentDocumentDatabase(...)))`

so you wouldn't need to override the `Dispose()`.